Update discussion of Version Negotiation (#4697)
quicwg/base-drafts: draft-ietf-quic-transport.md

retaining state.  Though either the Initial packet or the Version Negotiation
packet that is sent in response could be lost, the client will send new packets
until it successfully receives a response or it abandons the connection attempt.
As a result, the client discards all state for the connection and does not send

> any more packets on the connection.

Reviewer: Why remove this?  It doesn't seem to be responsive to the concerns that the rest of the change addresses.

Author: It's unrelated and could easily be elided.
I read through the whole top-level section while working on this PR, and I thought this looked out of place (it's the "Remove a sentence (apparently) about handling version negotiation packets from the section that's supposed to be about sending them.").  I could be mistaken about that, though, and we can split it out if that's preferred.

Reviewer: Please try to focus on targeted fixes.  That makes this process a lot easier to manage.

Author: Before I commit another faux pas, is it preferred to force-push or push a revert commit (or something else)?

Reviewer: I don't personally mind.  In this repo, we have let the commit history reflect the history of the issues.  So push as you like.

## Handling Version Negotiation Packets {#handle-vn}

Version Negotiation packets are designed to allow for functionality to be
defined in the future that allows QUIC to negotiate the version of QUIC to use

> between endpoints.  Future standards-track specifications might change how

Reviewer 2: ```suggestion
for a connection.  Future standards-track specifications might change how
```

packet that lists the QUIC version selected by the client.

How to perform version negotiation is left as future work defined by future
standards-track specifications.  In particular, that future work will
ensure robustness against version downgrade attacks; see
{{version-downgrade}}.  New versions of QUIC (including
non-standards-track versions) might be specified prior to the
availability of a version negotiation mechanism.  Until a version
negotiation mechanism is available, new versions of QUIC MUST respond to

> Version Negotiation packets as specified above.

Reviewer: I don't think that we need to include these last two sentences.  While we've definitely strayed into making forward-looking statements (read: predictions) about what we might do, this goes too far in my opinion.

Author: I concede it was a bit speculative to say that this reflects existing consensus.  That said, there is perhaps a spectrum of options, from removing it entirely to "are expected to", "SHOULD", etc.  It sounds like you prefer just removing it entirely, to confirm?

Reviewer: Yes.  I don't think that we get much value from saying anything like this, regardless of strength, simply because it is making a prediction about work that might happen in the future.

Author: Fair enough.  I will wait a bit before pushing an update to effectuate that, in case others want to comment.
